Lib/ldap/syncrepl.py

```diff
@@ -535,3 +536,71 @@ def syncrepl_refreshdone(self):
         follows.
         """
         pass
+
+
+class OpenLDAPSyncreplCookie:
+    """
+    OpenLDAPSyncreplCookie - allows a consumer to track a cookie across a
+    refreshAndPersist syncrepl session against a multi-provider OpenLDAP cluster
+    """
+
+    rid: int = 0
+    sid: int = 0
+    _csnset: Dict[int, str]
+
+    def __init__(self, cookie: AnyStr = "") -> None:
+        self._csnset = {}
+
+        if cookie:
+            self.update(cookie)
+
+    def _parse_csn(self, csn: str) -> List[str]:
```

Comment:
I think the return value can be a tuple here and we can specify the number of elements right away: ` -> Tuple[str, str, str, str]`

Also, it'll be nice to validate the tuple's len after the split.

Reply:
Good point, I'll adjust.